doc/rst/developer/bestPractices/Sanitizers.rst

```diff
@@ -0,0 +1,61 @@
+Sanitizers
+========
+
+How-to
+------
+
+To use AddressSanitizer with Chapel (compiler and executables):
```

Comment:
it may be just me, but I feel that AddressSanitizer is a much less common concept compared to valgrind. You may want to add a link to what little documentation is out there for it.

Reply:
In most of C/C++ land I think ASan is more well known. I'd be open to putting links in the documentation, but I feel like it's a term that's easy to google.

Reply:
I added some more information about sanitizers and asan to this file. I think it's enough to get started and we can add more later, but let me know if you'd like something else added now.

Reply:
Looks good